Implement some tests for request handler

andyg7 opened the pull request:
Hey guys,

1) I wrote a few tests here for request_handler.py, the new module I refactored out. As of now I just added tests for three methods from the RequestHandler class to get some feedback. I used MagicMock to mock out the api calls, and basically ran a few tests to make sure the proper code was called. Was looking just to get some code coverage going.

 2) I also simplified some of the tests from test_load_data.py as per the discussion from #110 that @Saturn suggested.

3) I also added mock==1.0.1 to the requirements.txt file.

4) As mentioned in a previous PR to run all tests run the following command: "python -m unittest discover tests" from the root of the repo. This will go into tests/ and look to run __all__ tests

Thanks for any feedback :)
Andy

carlosvargas commented on tests/test_request_handler.py:
```diff
@@ -0,0 +1,168 @@
+            self.rq._get(self.dummy_url)
+        except APIErrorException:
+            raised = True
+        self.assertTrue(raised is True)
```
For this and other response code tests, we should assert the message we output.

andyg7 commented:
Hey @carlosvargas,
Thanks a lot for the feedback. I replaced the raised checks with a assertFails in all cases, and also added some checks on the error messages that are reported when an exception is.
Let me know of any more feedback.
Thanks so much,
Andy

Saturn commented on tests/test_request_handler.py:
```diff
@@ -0,0 +1,162 @@
+        mock_writer.assert_called_once()
+
+    @mock.patch('test_request_handler.click.secho')
+    @mock.patch('test_request_handler.Stdout.team_scores')
```
Is there a reason this cannot be `@mock.patch('soccer.writer.Stdout.team_scores')` and `@mock.patch('click.secho')` etc?

andyg7 replied:
Ah good call thanks - looks like I can do what you mentioned. Will fix now.

andyg7 commented:
Hey guys, should I squash my commits whenever I make a change? I worry that re-writing history could mess things up (even though it's my forked version) but it also seems annoying to have a lot of commits with tiny changes.

carlosvargas commented:
I personally like to do that on PRs, but it's not a big deal.

Everything looks good to me 👍 Let's wait on @Saturn's input.

Saturn commented:
Looks good.
